Library/Homebrew/cmd/desc.rb

```diff
@@ -0,0 +1,38 @@
+require "descriptions"
+require "cmd/search"
+
+module Homebrew
+  def desc
+    if ARGV.options_only.empty?
+      if ARGV.named.empty?
+        raise FormulaUnspecifiedError
+        exit
+      end
+      results = Descriptions.named(ARGV.formulae.map(&:full_name))
+    else
+      searches = {
+        '-s' => :either,
+        '-n' => :name,
+        '-d' => :desc
```

Comment:
Added. (In the process, I replaced the somewhat convoluted Hash logic with a case statement, which should please @bfontaine. 😉)